uvicorn/config.py

```diff
@@ -446,6 +448,9 @@ def load(self) -> None:
                 ca_certs=self.ssl_ca_certs,
                 ciphers=self.ssl_ciphers,
             )
+
+        elif self.ssl_context:
+            self.ssl = self.ssl_context.custom_ssl_context_factory()  # type: ignore
```

Comment:
```suggestion
            self.ssl = self.ssl_context() 
```
this should do it no ? I dont get it otherwise

Reply:
it wont work with out calling the callable, here is my usage 
```
class SSL_Context(object):
    @classmethod
    def __call__(self):
        context = ssl.SSLContext(int(ssl.PROTOCOL_TLS))
        context.load_cert_chain(certfile=tls_cert, keyfile=tls_key)
        if allowed_ciphers:
            context.set_ciphers(allowed_ciphers)
        if list_options:
            for each_option in list_options:
                context.options |= each_option
        return context

ssl_context = SSL_Context

    uvicorn.run(
        "web:app",
        host="0.0.0.0",
        port=int(port),
        reload=True,
        ssl_context=ssl_context,
    )
```

and inside config.py 
```
        elif self.ssl_context:
            self.ssl = self.ssl_context.__call__()  # type: ignore
```